lib/stack_master/aws_driver/s3.rb

```diff
@@ -0,0 +1,73 @@
+require 'digest/md5'
+
+module StackMaster
+  module AwsDriver
+    class S3ConfigurationError < StandardError; end
+
+    class S3
+      def set_region(region)
+        @region = region
+        @s3 = nil
+      end
+
+      def upload_files(options)
+        bucket = options.fetch(:bucket) { raise StackMaster::AwsDriver::S3ConfigurationError, 'A bucket must be specified in order to use S3' }
+        prefix = options[:prefix]
+        bucket_region = options.fetch(:region, @region)
+        files = options.fetch(:files, [])
+
+        return if files.empty?
+
+        set_region(bucket_region) if bucket_region
+
+        current_objects = s3.list_objects(
+          prefix: prefix,
+          bucket: bucket
+        ).map(&:contents).flatten.inject({}){|h,obj|
+          h.merge(obj.key => obj)
+        }
+
+        files.each do |template, file|
+          body = file[:body]
+          file = file[:path]
+          key = template.dup
+          key.prepend("#{prefix}/") if prefix
+          file_md5 = Digest::MD5.file(file).to_s
+          body_md5 = Digest::MD5.hexdigest(body).to_s
+          s3_md5 = current_objects[key] ? current_objects[key].etag.gsub("\"", '') : nil
+
+          # require 'pry'
+          # binding.pry
+          unless file_md5 == s3_md5 or body_md5 == s3_md5
```

Comment:
While in this situation `or` works fine I would prefer to use `||`.

I'm not sure the file_md5 (file path hash) check is necessary here? Why would the s3 hash equal the hash of the file path locally?

Maybe this could be as simple as: `next if body_md5 == s3_md5` ?.